### PR TITLE
public sheet remapping

### LINKED ISSUE #2: Create needs components for each type of support
1. Financial resources
2. Grocery, meals, supplies, and errands (includes masks and sanitation equipment)
3. Ridesharing
4. Housing, storage, and petcare
5. In person childcare and online activities for kids
6. Emotional or spiritual support
7. Help filing for resources like unemployment, language resources, supports for undocumented folks, etc.
8. Specific items or other (i.e. doesn't fall into any category above)

### LINKED ISSUE #13: Show / hide needs based on visibility rules
We talked about this on the 8x8 call on 4/6/20 and what I understand is that we would hide entire submissions which are "No" for the following column. This means not generating any cards that would be parsed by the submission.
Screen Shot 2020-04-06 at 10 01 47 PM

There's also another column, which I'm not sure if / how we would use. Maybe @mewtaylor @mmoutenot have some better idea.
Screen Shot 2020-04-06 at 10 03 54 PM

### PR DESCRIPTION
**Pull request by esetnik**
1. public sheet remapping, resolves #13 
2. a new proposal for automatic metadata mapping, see https://github.com/mmoutenot/mamas-needs-offers/compare/master...esetnik:public-sheet-remap?expand=1#diff-635f2bf1b8b51f8413d1ef9d9d6360f5R40 for more info
3. update deps

Resolves #2, #3

### CODE REVIEW

File: src/utils/listingUtils.js [hunk 1]
```diff
@@ -1,44 +1,66 @@
...
-  isSuppliesNeed: 20,
-  supplies_needFrequency: 21,
-  supplies_needTiming: 22,
-  supplies_details: 23,
-  supplies_neighborhood: 24,
-  supplies_store: 25,
-  supplies_shoppingList: 26,
+  isSuppliesNeed: 18,
+  supplies_needFrequency: 19,
+  supplies_needTiming: 21,
```
Comment (mewtaylor) [inline src/utils/listingUtils.js:25]: There is also now one new question after timing, which is about covering cosots, so maybe we can call `supplies_costCoverage`

Comment (esetnik) [reply src/utils/listingUtils.js:25]: I missed that one! I'll update now.

Comment (esetnik) [reply src/utils/listingUtils.js:25]: @mewtaylor i'm a bit confused. Can you review the column headers for supplies on https://docs.google.com/spreadsheets/d/1TlDMhnDDuOAUzBTlW05t5mtoKUKcKp7Fmg8VL4PJOCU/edit#gid=2052572100 not sure which one is for the cost coverage.

Comment (mewtaylor) [reply src/utils/listingUtils.js:25]: Yes! it's the one that says "I would", V.

Comment (esetnik) [reply src/utils/listingUtils.js]: @mewtaylor should be fixed now

File: src/utils/listingUtils.js [hunk 2]
```diff
@@ -1,44 +1,101 @@
...
+      petComments: 58,
+    },
+  },
+
+  emotionalSupport: {
+    id: 59,
+    meta: {
+      supportType: 60,
+      resourceSupport: 61,
+      resourceSupportDetails: 62,
```
Comment (mewtaylor) [inline src/utils/listingUtils.js:87]: Ah @esetnik sorry one thing I missed! Resource support is its own category, after emotional support.

Comment (esetnik) [reply src/utils/listingUtils.js:87]: Ok so emotional support will only have one item with the support type?

Comment (mewtaylor) [reply src/utils/listingUtils.js:110]: Awesome, @esetnik you're the best

Comment (mewtaylor) [reply src/utils/listingUtils.js:110]: Love the new structure, too

**Comment by mewtaylor** [comment]
Looks great, I'd say go for the merge!

**Pull request merged**